### PR TITLE
added download function for geofence and rallypoints

### PR DESCRIPTION
**Pull request by mattes-bru**

### CODE REVIEW

File: protos/mission_raw/mission_raw.proto [hunk 1]
```diff
@@ -31,7 +31,15 @@ service MissionRawService {
     /*
      * Download a list of raw mission items from the system (asynchronous).
      */
-    rpc DownloadMission(DownloadMissionRequest) returns(DownloadMissionResponse) {}
+    rpc DownloadMission(DownloadMissionRequest) returns(DownloadMissionResponse) {}    
+    /*
+     * Download a list of raw mission items from the system (asynchronous).
+     */
+    rpc DownloadGeofence(DownloadMissionRequest) returns(DownloadMissionResponse) {}
```
Comment (julianoes) [inline protos/mission_raw/mission_raw.proto:38]:
```suggestion
    rpc DownloadGeofence(DownloadGeofenceRequest) returns(DownloadGeofenceResponse) {}
```

File: protos/mission_raw/mission_raw.proto [hunk 2]
```diff
@@ -31,7 +31,15 @@ service MissionRawService {
     /*
      * Download a list of raw mission items from the system (asynchronous).
      */
-    rpc DownloadMission(DownloadMissionRequest) returns(DownloadMissionResponse) {}
+    rpc DownloadMission(DownloadMissionRequest) returns(DownloadMissionResponse) {}    
+    /*
+     * Download a list of raw mission items from the system (asynchronous).
```
Comment (julianoes) [inline protos/mission_raw/mission_raw.proto:36]:
```suggestion
     * Download a list of raw geofence items from the system (asynchronous).
```

File: protos/mission_raw/mission_raw.proto [hunk 3]
```diff
@@ -31,7 +31,15 @@ service MissionRawService {
...
+     */
+    rpc DownloadGeofence(DownloadMissionRequest) returns(DownloadMissionResponse) {}
+        /*
+     * Download a list of raw mission items from the system (asynchronous).
```
Comment (julianoes) [inline protos/mission_raw/mission_raw.proto:40]:
```suggestion
     * Download a list of raw rally items from the system (asynchronous).
```

File: protos/mission_raw/mission_raw.proto [hunk 4]
```diff
@@ -31,7 +31,15 @@ service MissionRawService {
...
+     */
+     rpc DownloadRallypoints(DownloadMissionRequest) returns(DownloadMissionResponse) {}
```
Comment (julianoes) [inline protos/mission_raw/mission_raw.proto:42]:
```suggestion
     rpc DownloadRallypoints(DownloadRallypointsRequest) returns(DownloadRallypointsResponse) {}
```

**Review by julianoes — CHANGES_REQUESTED** [review]
See suggestions. The naming must be consistent, otherwise the auto-generation won't work. Best to test it before making the PR.

However, you probably can re-use the actual mission items in the requests, e.g.:

```
message DownloadGeofenceRequest {}
message DownloadGeofenceResponse {
    MissionRawResult mission_raw_result = 1;
    repeated MissionItem geofence_items = 2; // The geofence items
}
```

**Comment by mattes-bru** [comment]
Sorry, I was too fast on the PR button. I updated naming & comments

**Review by julianoes — APPROVED** [review]

**Pull request merged**